app/lib/actions/remote_execution/run_host_job.rb

```diff
@@ -38,9 +38,11 @@ def plan(job_invocation, host, template_invocation, proxy_selector = ::RemoteExe
 
         provider = template_invocation.template.provider
 
-        secrets = { :ssh_password => job_invocation.password || provider.ssh_password(host),
-                    :key_passphrase => job_invocation.key_passphrase || provider.ssh_key_passphrase(host),
-                    :sudo_password => job_invocation.sudo_password || provider.sudo_password(host) }
+        secrets = { :ssh_password => job_invocation.password,
+                    :key_passphrase => job_invocation.key_passphrase,
+                    :sudo_password => job_invocation.sudo_password }
+
+        secrets = secrets.merge(provider.secrets(host))
```

Comment:
This won't work if I am not mistaken. Previously the behaviour was "A | B" (A or B) and now it's just B because A is merged with B effectively overwriting the keys. What you probably want to do is to iterate over the secrets or use `map` call to keep the same behaviour. From the merge docs:

_If no block is given, entries with duplicate keys are overwritten with the values from each other_hash successively, otherwise the value for each duplicate key is determined by calling the block with the key, its value in the receiver and its value in each other_hash._

If this is what you want to do please explain then. In that case you can probably drop the merge and simply use the provider secrets instead?

Reply:
An oversight on my side, should be fixed now.